tests/ui.rs

```diff
@@ -338,6 +338,13 @@ fn main() -> Result<()> {
         ui(Mode::Fail, "tests/native-lib/fail", &target, WithoutDependencies, tmpdir.path())?;
     }
 
+    // We only enable GenMC tests when the `genmc` feature is enabled, but also only on platforms we support:
+    // FIXME(genmc,macos): Add `target_os = "macos"` once `https://github.com/dtolnay/cxx/issues/1535` is fixed.
+    if cfg!(all(feature = "genmc", target_os = "linux", target_pointer_width = "64")) {
+        ui(Mode::Pass, "tests/genmc/pass", &target, WithDependencies, tmpdir.path())?;
+        ui(Mode::Fail, "tests/genmc/fail", &target, WithDependencies, tmpdir.path())?;
+    }
```

Comment:
Remember that `cfg!` checks host conditions, even when it says `target_*`.
So this runs only on Linux hosts, but it runs on all targets, e.g. if I do `--target i686-unknown-linux-gnu`.

(a) you need logic somewhere that rejects unsupported targets when genmc is enabled -- must be 64bit little-endian
(b) here you need to avoid running tests on the wrong targets... the easiest thing is to check `host == target`. That's not great as then we don't do any cross-target checking, but that's fine for now.